Libraries/LibJS/Runtime/JSONObject.cpp

```diff
@@ -495,4 +503,60 @@ ThrowCompletionOr<Value> JSONObject::internalize_json_property(VM& vm, Object* h
     return TRY(call(vm, reviver, holder, PrimitiveString::create(vm, name.to_string()), value));
 }
 
+// https://tc39.es/proposal-json-parse-with-source/#sec-json.rawjson
+JS_DEFINE_NATIVE_FUNCTION(JSONObject::raw_json)
+{
+    auto& realm = *vm.current_realm();
+
+    // 1. Let jsonString be ? ToString(text).
+    auto json_string = TRY(vm.argument(0).to_string(vm));
+
+    // 2. Throw a SyntaxError exception if jsonString is the empty String, or if either the
+    // first or last code unit of jsonString is any of 0x0009 (CHARACTER TABULATION),
+    // 0x000A (LINE FEED), 0x000D (CARRIAGE RETURN), or 0x0020 (SPACE).
+    auto bytes = json_string.bytes_as_string_view();
+    if (bytes.is_empty())
+        return vm.throw_completion<SyntaxError>(ErrorType::JsonMalformed);
+
+    auto first_char = bytes[0];
+    auto last_char = bytes[bytes.length() - 1];
+
+    if (first_char == 0x0009 || first_char == 0x000A || first_char == 0x000D || first_char == 0x0020 || last_char == 0x0009 || last_char == 0x000A || last_char == 0x000D || last_char == 0x0020)
+        return vm.throw_completion<SyntaxError>(ErrorType::JsonMalformed);
```

Comment:
This line is a bit beefy - I'd maybe define the code points in a span and then check the span for the first/last char in the string. Something like:

```c++
static constexpr AK::Array invalid_code_points { 0x09, 0x0A, 0x0D, 0x20 };
auto first = bytes[0];
auto last = bytes[bytes.length() - 1];

if (invalid_code_points.contains_slow(first) || invalid_code_points.contains_slow(last))
    return vm.throw_completion<SyntaxError>(ErrorType::JsonMalformed);
```

Reply:
Oh, that's a nice way to do this, thank you!